python/ray/tests/test_object_spilling.py

```diff
@@ -229,6 +230,18 @@ def task():
     ray.get([task.remote() for _ in range(2)])
 
 
+def test_custom_spill_dir(shutdown_only):
+    # Make sure the object spilling storage path can be set by the user
+    ray.init(object_spilling_storage_path="/tmp/custom_spill_dir")
+    config = json.loads(
+        ray._private.worker._global_node._config["object_spilling_config"]
+    )
+    assert config["type"] == "filesystem"
+    assert config["params"]["directory_path"] == "/tmp/custom_spill_dir"
```

Comment:
better to write a test that actually validates the behavior instead of one that calls into private APIs

for example, set object store memory very low and check that files are created in the spilling dir